### PR TITLE
add clearSelect/Where/OrderBy/Limit/Offset

### PR DESCRIPTION
**Pull request by wirekang**
Resetting part of the query is a useful feature. It's my first contribution for this project, so please point out any problems.

Related #249 

todo
 * [x] implement.
 * [x] unit tests.
 * [x] typing tests.
 * [x] docs.

### CODE REVIEW

**Comment by wirekang** [comment]
I think I've done all the work(except documentation). Please let me know if there is a problem.

**Comment by koskimas** [comment]
Looking good! Please install prettier and run it for all the files you've changed though.

**Comment by wirekang** [comment]
I ran preitter, and added ts docs. Please check it out.

File: src/query-builder/select-query-builder.ts [hunk 1]
```diff
@@ -1464,6 +1464,132 @@ export class SelectQueryBuilder<DB, TB extends keyof DB, O>
...
+   *   .clearWhere()
+   * ```
+   *
+   * The generated SQL(PostgreSQL):
+   *
+   * ```sql
+   * select * from "person"
+   * ```
+   */
+  clearWhere(): SelectQueryBuilder<DB, TB, O> {
```
Comment (koskimas) [inline src/query-builder/select-query-builder.ts:1510]: Move this doc to `WhereInterface`. We have a script that copies the comment to the implementing method after build.

File: test/node/src/clear.test.ts [hunk 2]
```diff
@@ -0,0 +1,192 @@
+import { BUILT_IN_DIALECTS, initTest, TestContext, testSql } from './test-setup'
```
Comment (koskimas) [inline test/node/src/clear.test.ts:1]: Run prettier for this file

Comment (koskimas) [reply test/node/src/clear.test.ts:1]: We should add a linter

Comment (wirekang) [reply test/node/src/clear.test.ts:1]: `npm run prettier` does not format this file. I don't know why, but I do it manually anyway.

**Comment by wirekang** [comment]
Thank you for your kind feedback.

**Comment by koskimas** [comment]
Thank you! I'll release a new version soon.

**Pull request merged**